test/python/test_identifiers.py

```diff
@@ -306,1107 +308,252 @@ def test_add_circuit_noname(self):
-        target = {'00': shots / 2, '01': shots / 2}
-        threshold = 0.04 * shots
-        self.assertDictAlmostEqual(counts, target, threshold)
-
-
-class TestAnonymousIdsNoQuantumProgram(QiskitTestCase):
-    """Test the anonymous use of registers.
-    TODO: this needs to be expanded, ending up with the rest of the tests
-    in the file not using QuantumProgram when it is deprecated.
+def test_invalid_qasmname_qr_spec(self):
+    """QPS_SPECS_NONAMES defines a quantum register with invalid QASM name (do not start
+    with lowercase).
     """
-
-    def test_create_anonymous_classical_register(self):
-        """Test creating a ClassicalRegister with no name.
-        """
-        cr = ClassicalRegister(size=3)
-        self.assertIsInstance(cr, ClassicalRegister)
-
-    def test_create_anonymous_quantum_register(self):
-        """Test creating a QuantumRegister with no name.
-        """
-        qr = QuantumRegister(size=3)
-        self.assertIsInstance(qr, QuantumRegister)
-
-    def test_create_anonymous_classical_registers(self):
-        """Test creating several ClassicalRegister with no name.
-        """
-        cr1 = ClassicalRegister(size=3)
-        cr2 = ClassicalRegister(size=3)
-        self.assertNotEqual(cr1.name, cr2.name)
-
-    def test_create_anonymous_quantum_registers(self):
-        """Test creating several QuantumRegister with no name.
-        """
-        qr1 = QuantumRegister(size=3)
-        qr2 = QuantumRegister(size=3)
-        self.assertNotEqual(qr1.name, qr2.name)
-
-    def test_create_anonymous_mixed_registers(self):
-        """Test creating several Registers with no name.
-        """
-        cr0 = ClassicalRegister(size=3)
-        qr0 = QuantumRegister(size=3)
-        # Get the current index counte of the registers
-        cr_index = int(cr0.name[1:])
-        qr_index = int(qr0.name[1:])
-
-        cr1 = ClassicalRegister(size=3)
-        _ = QuantumRegister(size=3)
-        qr2 = QuantumRegister(size=3)
-
-        # Check that the counters for each kind are incremented separately.
-        cr_current = int(cr1.name[1:])
-        qr_current = int(qr2.name[1:])
-        self.assertEqual(cr_current, cr_index + 1)
-        self.assertEqual(qr_current, qr_index + 2)
-
-    def test_create_circuit_noname(self):
-        """Test create_circuit with no name
-        """
-        q_program = QuantumProgram()
-        qr = QuantumRegister(size=3)
-        cr = ClassicalRegister(size=3)
-        qc = q_program.create_circuit(qregisters=[qr], cregisters=[cr])
-        self.assertIsInstance(qc, QuantumCircuit)
+    QPS_SPECS_NONAMES = {
+        "circuits": [{
+            "quantum_registers": [{
+                "name": 'Qr',
+                "size": 3}],
+            "classical_registers": [{
+                "size": 3}]
+        }]
+    }
+
+    self.assertRaises(QISKitError, QuantumProgram, specs=QPS_SPECS_NONAMES)
+
+
+def test_invalid_qasmname_cr_spec(self):
```

Comment:
Can you please move all the `test_something` to inside a `TestCase`? https://ci.appveyor.com/project/QISKit/qiskit-sdk-py/build/1.0.834/job/jy0qec4n3yukkusk/tests

Reply:
oh.. of course. Probably a bad merge/copypaste/something. Fixed in 
e01099b